src/edu/csus/ecs/pc2/ui/JListPane.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.ui;
+
+import java.awt.BorderLayout;
+import java.awt.FlowLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+
+/**
+ * A generic list pane
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class JListPane extends JPanePlugin {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -1383088589409036086L;
+
+	@SuppressWarnings("rawtypes")
+	JList theList = new JList();
+
+	private JFrame parentFrame;
+
+	private int[] holdSelectedIndexes = new int[0];
+
+	private ISelectedListsSetter selectedListsSetter;
+
+	public JListPane() {
```

Comment:
Class constructors should have header comments explaining what they do -- especially when the class has more than one constructor like this class does (the header comments should, among other things, make it clear what the difference between constructors is...)

Reply:
Also:  if someone calls this constructor directly, they will get a `JListPane` that contains a SAVE button with an `ActionListener` which calls `saveAndClose()`.  The `JListPane` will not have a "ParentFrame" (because that only gets created if you call the _other_ constructor).  If someone then hits "Save", the call to `parentFrame.setVisible(false);` in method `saveAndClose()` will throw a NullPointerException.  That needs to be fixed.

Reply:
Will fix NPE bugs